lib/matplotlib/dviread.py

```diff
@@ -864,100 +870,83 @@ def __getitem__(self, texname):
                    'This problem can often be solved by installing '
                    'a suitable PostScript font package in your (TeX) '
                    'package manager.')
-            msg = fmt.format(texname.decode('ascii'), self._filename)
-            msg = textwrap.fill(msg, break_on_hyphens=False,
-                                break_long_words=False)
-            _log.info(msg)
+            _log.info(textwrap.fill(
+                fmt.format(texname.decode('ascii'), self._filename),
+                break_on_hyphens=False, break_long_words=False))
             raise
-        fn, enc = result.filename, result.encoding
-        if fn is not None and not fn.startswith(b'/'):
-            fn = find_tex_file(fn)
-        if enc is not None and not enc.startswith(b'/'):
-            enc = find_tex_file(result.encoding)
-        return result._replace(filename=fn, encoding=enc)
-
-    def _parse(self, file):
-        """
-        Parse the font mapping file.
-
-        The format is, AFAIK: texname fontname [effects and filenames]
-        Effects are PostScript snippets like ".177 SlantFont",
-        filenames begin with one or two less-than signs. A filename
-        ending in enc is an encoding file, other filenames are font
-        files. This can be overridden with a left bracket: <[foobar
-        indicates an encoding file named foobar.
 
-        There is some difference between <foo.pfb and <<bar.pfb in
-        subsetting, but I have no example of << in my TeX installation.
+    def _parse_and_cache_line(self, line):
+        """
+        Parse a line in the font mapping file.
+
+        The format is (partially) documented at
+        http://mirrors.ctan.org/systems/doc/pdftex/manual/pdftex-a.pdf
+        https://tug.org/texinfohtml/dvips.html#psfonts_002emap
+        Each line can have the following fields:
+
+        - tfmname (first, only required field),
+        - psname (defaults to tfmname, must come immediately after tfmname if
+          present),
+        - fontflags (integer, must come immediately after psname if present,
+          ignored by us),
+        - special (SlantFont and ExtendFont, only field that is double-quoted),
+        - fontfile, encodingfile (optional, prefixed by <, <<, or <[ possibly
+          with an space after the prefix; << always precedes a font, <[ always
```

Comment:
Is space optional after any prefix, or just `<`? I only see space after `<` in the test file, and I don't see it handled, so I think it's the latter.

Reply:
https://tug.org/texinfohtml/dvips.html#psfonts_002emap states

> Otherwise, the line is separated into words, where words are separated by spaces or tabs, except that if a word begins with a double quote, it extends until the next double quote or the end of the line.
> If a word starts with ‘<<’, it is taken as a font file to be wholly downloaded. Use this to avoid partial downloading, as described above.
> If a word starts with ‘<[’, it is taken as an encoding file to be downloaded. Use this if the name of the encoding file does end in ‘.enc’, also as described above.
> If a word starts with a ‘<’ character, it is treated as a header file that needs to be downloaded. If the name ends in ‘.pfa’ or ‘.pfb’, it is taken as Type 1 font file that will be partially downloaded if the ‘j’ option is in effect. There can be more than one such header for a given font. If a ‘<’ is a word by itself, the next word is taken as the name of the header file.

which I understand as meaning that *only* < can be followed by a space.

Reply:
OK, but the comment makes it seem like space can be after any of them, or possibly just `<[`.

Reply:
ah, reworded